Cookbook/Technical-Documents/CRP-Bot.md

```diff
@@ -0,0 +1,101 @@
+
+> Note: for documentation about how the CRP is _implemented_ in the code of our Stevenson bot, see [this companion document](CRP-Bot-ImplementationDetails.md)._
+
+## Projects whitelist
+
+To avoid accidentally creating JIRA versions on boards others than the ones we own, or messing with boards that don't follow the same schema/template as the one expected by the bot, the CRP bot has a whitelist of boards on which it will be allowed to create JIRA versions and set Fix Versions fields.
+
+* This provides some security against any `[XXX-NNN]` that might be contained in a commit message to reference a JIRA ticket on some board that we don't own (e.g. a ticket from BackEnd), for which we don't want our bot to act
+* This also allows us to not whitelist some of our boards, for which we ideally would want the CRP bot to work on, but which sadly don't use the official JIRA template that the bot expects – which means that they might not have the same fields in their JIRA tickets, or have different field IDs in the JIRA API for some fields (and sending requests to update a given Field ID would mean accidentally update the wrong field)
+
+This is why some of the boards are still not whitelisted. This is tracked in [IOSP-101](https://babylonpartners.atlassian.net/browse/IOSP-101)
+
+Note that the whitelist only affects the creation of JIRA releases and update of "Fix Version" field on those boards. A link to the tickets will still be included in the description/body of CRP ticket (listing all tickets found in the changelog messages) even for tickets belonging to a non-whitelisted board.
+
+## Running the CRP bot
+
+Currently, the CRP bot can be triggered by a slack command `/crp <platform> <branch>` (e.g. `/crp ios release/babylon/4.4.0`) in the `#ios-launchpad` slack channel.
+
+In the near future, it will likely be automatically triggered by the script doing the release cut every second Friday.
+
+## JIRA API limitations
+
+Given some quota limitations with the JIRA API – which makes JIRA not respond to some requests if we send too many requests in parallel – the CRP bot uses some throttling mechanism to only make API calls to set the "Fix Version" field of each ticket by batches, instead of all at once. This improves the behavior of the JIRA API, but that API still sometimes fail with timeout on some requests – often the JIRA server seem to have processed the request and updated the Fix Version field as requested by the API call, but never returns an HTTP response to the client to let it know it succeeded.
+
+Currently, since the default timeout for network requests is 60s, every time JIRA decides not to respond to one of our API requests, that request takes 60s to finally fail. This means that on those occasions, the CRP process can take as much as N minutes where N is the number of timeout failures on which JIRA API decided not to reply to our requests.  
```

Comment:
```suggestion
As of January 2020, since the default timeout for network requests is 60s, every time JIRA decides not to respond to one of our API requests, that request takes 60s to finally fail. This means that on those occasions, the CRP process can take as much as N minutes where N is the number of timeout failures on which JIRA API decided not to reply to our requests.  
```